tools/tplg_parser/tokens.c

```diff
@@ -49,7 +49,8 @@ int get_token_uint32_t(void *elem, void *object, uint32_t offset,
 		       uint32_t size)
 {
 	struct snd_soc_tplg_vendor_value_elem *velem = elem;
-	uint32_t *val = object + offset;
+	char *vobject = object;
+	uint32_t *val = (uint32_t *)(vobject + offset);
```

Comment:
So `sizeof(char)` is always 1. If a `char` had say 16 bits, then `sizeof(uint8_t)` would be... 0.5? 1? Undefined?

`sizeof(uint32_t)` would likely be 2 and I bet a lot of other code would break :-)

Reply:
@marc-hb I think on platforms where char is 16bit, there is no uint8_t, so sizeof(uint8_t) would not be a problem. But yeah, I think POSIX guarantees char to be 8bits, so this is definitely beyond academic for testbench.